### ISSUE TITLE
`Concatenate` cannot be used to add or remove arguments.

### ISSUE THREAD

**Issue by ramcasa** [astral-sh/ty#2774]
### Summary

Some basic examples from the [python docs] are failing. The decorated function types below are not correct and one even gives an error.

```python
from collections.abc import Callable
from threading import Lock
from typing import ParamSpec, Concatenate
from typing_extensions import reveal_type

Signature = ParamSpec("Signature")

my_lock = Lock()


def pass_lock(
    f: Callable[Concatenate[Lock, Signature], int],
) -> Callable[Signature, int]:
    """A type-safe decorator which provides a lock."""

    def inner(*args: Signature.args, **kwargs: Signature.kwargs) -> int:
        # Signaturerovide the lock as the first argument.
        return f(my_lock, *args, **kwargs)

    return inner


@pass_lock
def sum_threadsafe(lock: Lock, numbers: list[float]) -> int:
    """Add a list of numbers together in a thread-safe manner."""
    with lock:
        return int(sum(numbers))


reveal_type(sum_threadsafe)


def with_lock(
    f: Callable[Signature, int],
) -> Callable[Concatenate[Lock, Signature], int]:
    """A type-safe decorator which provides a lock."""

    def inner(lock: Lock, *args: Signature.args, **kwargs: Signature.kwargs) -> int:
        # Signaturerovide the lock as the first argument.
        with lock:
            return f(*args, **kwargs)

    return inner


@with_lock
def sum_threadunsafe(numbers: list[float]) -> int:
    """Add a list of numbers together in a thread-safe manner."""
    return int(sum(numbers))

reveal_type(sum_threadunsafe)

sum_threadsafe([1, 2, 3])
sum_threadunsafe([1, 2, 3])
```

<details>
<summary>Check Output</summary>

```
info[revealed-type]: Revealed type
  --> report_ty_generic_decorator_factories.py:30:13
   |
30 | reveal_type(sum_threadsafe)
   |             ^^^^^^^^^^^^^^ `[**Signature'return](**Signature'return@pass_lock) -> int`
   |

info[revealed-type]: Revealed type
  --> report_ty_generic_decorator_factories.py:51:13
   |
49 |     return int(sum(numbers))
50 |
51 | reveal_type(sum_threadunsafe)
   |             ^^^^^^^^^^^^^^^^ `(...) -> int`
52 |
53 | sum_threadsafe([1, 2, 3])
   |

error[missing-argument]: No argument provided for required parameter `**kwargs`
  --> report_ty_generic_decorator_factories.py:53:1
   |
51 | reveal_type(sum_threadunsafe)
52 |
53 | sum_threadsafe([1, 2, 3])
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
54 | sum_threadunsafe([1, 2, 3])
   |
info: These arguments are required because `ParamSpec` `Signature'return` could represent any set of parameters at runtime
info: rule `missing-argument` is enabled by default

Found 3 diagnostics
```

</details>

[python docs]: https://docs.python.org/3/library/typing.html#typing.Concatenate

### Version

ty 0.0.15

**Comment by ramcasa** [comment]
@AlexWaygood, by the name of the issue and the link in it, it seemed to me #1535 is talking about supporting the syntax `Concatenate[whatever, P]` where `P` is specified before as `[**P]` instead of using a `ParamSpec` variable. Is it not? I suggest renaming the issue to `Support Concatenate` if not. (Or is there some form that works?)

**Comment by AlexWaygood** [comment]
@ramcasa, we don't support any use of `Concatenate` at all at the moment; #1535 is an umbrella issue covering that :-)
